Superchain Registry docs (#1125)
ethereum-optimism/docs: pages/superchain/superchain-registry.mdx

---
title: The Superchain Registry
lang: en-US

> description: Learn about Optimism Superchain components, features, and roadmap.

Reviewer: ```suggestion
description: Learn about the members of the Optimism Superchain ecosystem.
```

---

import { Callout } from 'nextra/components'

# The Superchain Registry

The Superchain Registry serves as the source of truth for who's in the Superchain Ecosystem and what modifications they've made. The Superchain Registry introduces:

> *   A step-by-step process new chains can follow to join the registry

Reviewer: ```suggestion
*   A step-by-step process new chains can follow to join the Registry
```

*   Validation checks to ensure standard chains comply with the Standard Rollup Charter and non-standard chains pass baseline validation before joining the Superchain Registry

> *   A Superchain Registry repository that shows who's in the registry and the chain's configuration. Chains with `superchain_level = 1` meet all the criteria to be classified as a Standard Rollup.

Reviewer: ```suggestion
*   A Superchain Registry repository that shows who's in the registry and the chain's configuration. Chains with `superchain_level = 1` have proven that they follow the Standard Rollup Charter, and can be classified as a Standard Rollup.
```

## The Standard Rollup

> An OP Stack Standard Rollup meets the set of requirements to be a member of the Superchain.

Reviewer: ```suggestion
An OP Stack Standard Rollup follows the Standard Rollup Charter.
```

> All chains must meet this config to meet the inclusion requirements for The Superchain. This configuration targets the Optimism Collective's highest bar for security, uptime, and decentralization.

Reviewer: ```suggestion
This configuration targets the Optimism Collective's highest bar for security, uptime, and decentralization.
```

You can find more details in the [Standard Rollup Charter documentation](/superchain/blockspace-charter).

<Callout type="info">

>   We recommend using the [op-deployer](/builders/chain-operators/tools/op-deployer) to deploy L1 contracts and generate the L2 genesis file that meet the configuration requirements outlined in the [Standard Rollup Charter](/superchain/blockspace-charter).

Reviewer: ```suggestion
  We **strongly** recommend using the [op-deployer](/builders/chain-operators/tools/op-deployer) to deploy L1 contracts and generate the L2 genesis file that meet the configuration requirements outlined in the [Standard Rollup Charter](/superchain/blockspace-charter).
```

</Callout>

> ## Joining the registry

Reviewer: ```suggestion
## Joining the Registry
```

---
title: The Superchain Registry
lang: en-US
description: Learn about Optimism Superchain components, features, and roadmap.
---

import { Callout } from 'nextra/components'

# The Superchain Registry

> The Superchain Registry is a human-readable index of chains which serves as the source of truth for who's in the Superchain and what modifications they've made. The Superchain Registry introduces:

Reviewer: ```suggestion
The Superchain Registry serves as the source of truth for who's in the Superchain Ecosystem and what modifications they've made. The Superchain Registry introduces:
```

We **must** always use the term Superchain Ecosystem rather than just Superchain here.

> *   A clear definition of what constitutes a Standard Chain and config

Reviewer 2: ```suggestion
```

We actually don't do this in the registry in a clear way

*   A step-by-step process new chains can follow to join the registry

> *   Validation checks to ensure chains follow the standard config before joining the Superchain Registry

Reviewer 2: ```suggestion
*   Validation checks to ensure standard chains comply with the Standard Rollup Charter and non-standard chains pass baseline validation before joining the Superchain Registry
```

> *   A Superchain Registry repository that shows who's in the registry and what changes they have made to the standard config

Reviewer 2: ```suggestion
*   A Superchain Registry repository that shows who's in the registry and the chain's configuration
```

You can find the [Superchain Registry here](https://github.com/ethereum-optimism/superchain-registry)

> ## The Standard Config

Reviewer 2: ```suggestion
## The Standard Rollup
```

I know we use "Standard Config" a lot, but I think we should move towards language defined in the Standard Rollup Charter: https://github.com/ethereum-optimism/OPerating-manual/blob/feat/standard-rollup-charter/Standard%20Rollup%20Charter.md#standard-rollup-charter

Author: Gotcha, I'll remove it from these docs and take a pass at other docs like this in the future: https://docs.optimism.io/builders/chain-operators/configuration/rollup

> The Standard Config is the set of requirements for an OP Stack chain to be considered a Standard Chain within the Superchain. These requirements are currently a draft, pending governance approval.

Reviewer 2: ```suggestion
An OP Stack Standard Rollup meets the set of requirements to be a member of the Superchain. 
```

> All chains must meet this config to meet the inclusion requirements for The Superchain. This config isn't meant to exclude chains, but instead make it easier to understand *how* to configure your chain to make it compatible with the Superchain.

Reviewer 2: ```suggestion
All chains must meet this config to meet the inclusion requirements for The Superchain. This configuration targets the Optimism Collective’s highest bar for security, uptime, and decentralization.
```

> You can find the current draft version of The Standard Config in the [specs repo](https://specs.optimism.io/protocol/configurability.html#op-stack-configurability).

Reviewer 2: ```suggestion
You can find more details in the [Standard Rollup Charter documentation](/superchain/blockspace-charter).
```

<Callout type="info">

>   We recommend using the [OP Contract Manager](/stack/opcm) to deploy L1 contracts that meet the Standard Config.

Reviewer 2: ```suggestion
  We recommend using the [op-deployer](/builders/chain-operators/tools/op-deployer) to deploy L1 contracts and generate the L2 genesis file that meet the configuration requirements outlined in the [Standard Rollup Charter](/superchain/blockspace-charter).
```

</Callout>

## Joining the registry

> The Superchain Registry is designed to outline and clarify the process for making your chain Superchain-compatible and joining the registry.

Reviewer: ```suggestion

```

Recommend deleting this line

> If your chain meets the Standard Config requirements, [follow the steps in this guide](https://github.com/ethereum-optimism/superchain-registry/blob/main/docs/add-chain.md) to submit a pull request and join the Superchain. This guide will help you set up your chain, create your environment, and run all the validation checks needed to ensure your chain is ready to join the registry.

Reviewer: ```suggestion
All Superchain Ecosystem members are welcome to join the Superchain Registry, regardless of whether they adhere to the standard rollup charter to not. To join the Registry, [follow the steps in this guide](https://github.com/ethereum-optimism/superchain-registry/blob/main/docs/add-chain.md) to submit a pull request and join the Superchain Ecosystem. This guide will help you set up your chain, create your environment, and run all the validation checks needed to ensure your chain is ready to join the Registry.
```
